src/mlpack/core/data/tokenizers/split_by_any_of.hpp

```diff
@@ -51,17 +51,16 @@ class SplitByAnyOf
    *
    * @param str String view to retrieve the token from.
    */
-  boost::string_view operator()(boost::string_view& str) const
+  STRING_VIEW operator()(STRING_VIEW& str) const
   {
-    boost::string_view retval;
+    STRING_VIEW retval;
 
     while (retval.empty())
     {
       const std::size_t pos = FindFirstDelimiter(str);
       if (pos == str.npos)
       {
         retval = str;
-        str.clear();
```

Comment:
@shrit did you mean to remove this line?  I read through the debugging (nice digging by the way!) and it seems to me like this line being removed could be the problem.  If I am reading this right, this means that the input `str` is returned as `retval` if we can't find any more delimiters.

Reply:
Oh, Yeah I remember removing this line, there was an error caused by this line, I do not remember what it was.
I will put it back and see what was the error.

Reply:
I remember now why I have removed the `str.clear()` line. Since the `str` is a string_view object, the `clear()` function is not part of the STL, but instead, it is a part of the `boost::string_view` which is strange by the way, making the Boost API not compatible with the `STL` API.
We need to find another way to clear the string view.

Reply:
Ah, ok---I think one way or another all we need to do here is make sure that `str` points to an empty string.  So maybe we can do that in the same way as the other branch, using `remove_prefix()` or similar?